src/util/Error.h

```diff
@@ -74,8 +74,11 @@ struct ErrorCode {
  */
 class Error {
 public:
-    Error(int code = ErrorCode::NONE, std::string const& msg = "", int status = ErrorCode::NONE,
-          bool logLvLErr = true);
+    Error(int code, std::string const& msg = "", int status = ErrorCode::NONE, bool logLvLErr = true);
```

Comment:
You may want to add `explicit`:
```
  explicit Error(int code, std::string const& msg = "", int status = ErrorCode::NONE, bool logLvLErr = true);
```
Normally, you use `explicit` in all cases where you have a single parameter constructor because C++ may use the constructor as a type conversion operator. In this particular case, if you has some method accepting:
```
void foo(Error err) {
...
```
Then the conversion constructor would let you do:
```
foo(123);
```
Which would be the same as:
```
foo(Error(123));
```
You may not want this ... unless you rely on such a form of type conversion on purpose. These conversions may be a dangerous thing.

Reply:
Good point.